tests/everest/test_everserver.py

```diff
@@ -191,24 +252,37 @@ def test_everserver_status_exception(
     # The server should fail, and store the exception that
     # start_optimization raised.
     assert status["status"] == ServerStatus.failed
-    assert "Exception: Failed optimization" in status["message"]
+    assert "Optimization failed with exception." in status["message"]
 
 
+@pytest.mark.integration_test
+@pytest.mark.xdist_group(name="starts_everest")
+@pytest.mark.timeout(120)
 @patch("sys.argv", ["name", "--config-file", "config_minimal.yml"])
-@patch(
-    "everest.detached.jobs.everserver._sim_monitor",
-    side_effect=partial(set_shared_status, progress=[]),
-)
-def test_everserver_status_max_batch_num(
-    _1, mock_server, copy_math_func_test_data_to_tmp
-):
+async def test_status_max_batch_num(copy_math_func_test_data_to_tmp):
     config = EverestConfig.load_file("config_minimal.yml")
     config.optimization = OptimizationConfig(
         algorithm="optpp_q_newton", max_batch_num=1
     )
     config.dump("config_minimal.yml")
 
-    everserver.main()
+    async def server_running():
+        while True:
+            event = await driver.event_queue.get()
+            if isinstance(event, FinishedEvent) and event.iens == 0:
+                return
+
+    driver = await start_server(config, debug=True)
+    try:
+        wait_for_server(config.output_dir, 120)
+        start_experiment(
+            server_context=ServerConfig.get_server_context(config.output_dir),
+            config=config,
+        )
+    except (SystemExit, RuntimeError) as e:
+        raise e
+    await server_running()
```

Comment:
Perhaps make this a small function for reuse? Looks like it can be used in the test below as well?